uikit/form/MultiSelect/index.tsx

```diff
@@ -334,7 +334,9 @@ const MultiSelect = ({
   };
 
   const handleInputChange = (event) => {
-    setSearchString(event.target.value);
+    const newValue = event.target.value;
+    setSearchString(newValue);
+    single && onChange(createCustomEvent(event, [newValue]), null);
```

Comment:
If this is a single selection, there's no reason why the value should be an array.
If an array output is needed at integration, it can be added on the receiving end (e.g. input handler).

```suggestion
    single && onChange(createCustomEvent(event, newValue), null);
```

Reply:
discussed on slack: this likely has to be an array, see `handleNewItemClick` above

Reply:
Confirmed. The pattern is not ideal (`single` vs `multiple` should produce clearly different results), but improving it would require additional work in Platform, etc.

i.e. to be improved upon at a later point.